js/search.js

```diff
@@ -927,6 +927,7 @@
 			return this.renderPokemonSortRow();
 		case 'pokemon':
 			var pokemon = BattlePokedex[id];
+			if (!pokemon) pokemon = BattlePokedex[toId(Tools.getTemplate(id).baseSpecies)];
```

Comment:
This becomes a case only now, since forms aren't entries in `BattlePokedex` unlike everything else but shouldn't break the search.